src/client/app/components/meters/MetersDetailComponent.tsx

```diff
@@ -4,129 +4,71 @@
-	// current user state
-	const currentUserState = useSelector((state: State) => state.currentUser);
-
 	// Check for admin status
-	const currentUser = currentUserState.profile;
-	const loggedInAsAdmin = (currentUser !== null) && isRoleAdmin(currentUser.role);
-
+	const isAdmin = useAppSelector(selectIsAdmin);
 	// We only want displayable meters if non-admins because they still have
 	// non-displayable in state.
-	let visibleMeters;
-	if (loggedInAsAdmin) {
-		visibleMeters = MetersState;
-	} else {
-		visibleMeters = _.filter(MetersState, (meter: MeterData) => {
-			return meter.displayable === true
-		});
-	}
-
-	// Units state
-	const units = useSelector((state: State) => state.units.units);
-	// Units state loaded status
-	const unitsStateLoaded = useSelector((state: State) => state.units.hasBeenFetchedOnce);
-
-	// Possible Meter Units to use
-	let possibleMeterUnits = new Set<UnitData>();
-	// The meter unit can be any unit of type meter.
-	Object.values(units).forEach(unit => {
-		if (unit.typeOfUnit == UnitType.meter) {
-			possibleMeterUnits.add(unit);
-		}
-	});
-	// Put in alphabetical order.
-	possibleMeterUnits = new Set(_.sortBy(Array.from(possibleMeterUnits), unit => unit.identifier.toLowerCase(), 'asc'));
-	// The default graphic unit can also be no unit/-99 but that is not desired so put last in list.
-	possibleMeterUnits.add(noUnitTranslated());
-
-	// Possible graphic units to use
-	const possibleGraphicUnits = potentialGraphicUnits(units);
-
-	const titleStyle: React.CSSProperties = {
-		textAlign: 'center'
-	};
-
-	const tooltipStyle = {
-		display: 'inline-block',
-		fontSize: '50%',
-		// Switch help depending if admin or not.
-		tooltipMeterView: loggedInAsAdmin ? 'help.admin.meterview' : 'help.meters.meterview'
-	};
+	const { visibleMeters } = useAppSelector(selectVisibleMeterAndGroupData);
 
 	return (
 		<div>
-			<HeaderComponent />
-			<TooltipHelpContainer page='meters' />
+			<TooltipHelpComponent page='meters' />
 
 			<div className='container-fluid'>
 				<h2 style={titleStyle}>
 					<FormattedMessage id='meters' />
 					<div style={tooltipStyle}>
-						<TooltipMarkerComponent page='meters' helpTextId={tooltipStyle.tooltipMeterView} />
+						<TooltipMarkerComponent page='meters' helpTextId={getToolTipMessage(isAdmin)} />
 					</div>
 				</h2>
-				{loggedInAsAdmin && metersStateLoaded && unitsStateLoaded &&
+				{isAdmin &&
 					<div className="edit-btn">
-						{/* The actual button for create is inside this component. */}
-						<CreateMeterModalComponent
-							possibleMeterUnits={possibleMeterUnits}
-							possibleGraphicUnits={possibleGraphicUnits}
-						/>
+						<CreateMeterModalComponent />
 					</div>
 				}
-				{metersStateLoaded && unitsStateLoaded &&
+				{
 					<div className="card-container">
 						{/* Create a MeterViewComponent for each MeterData in Meters State after sorting by identifier */}
+						{/*  Optional Chaining to prevent from crashing upon startup race conditions*/}
```

Comment:
Can you help me understand why this is optional chaining?

Reply:
This comment should be removed. It seems that this comment is still lingering around after some of the logic was shifted elsewhere.. This optional chaining still occurs in many of the selectors. 
One issue I was encountering during was that selectors were firing before the initial data was loading. This would cause fatal undefined index access issues. To prevent this I utilized optional chaining to short-circut some of the methods. Doing this caused the methods to gracefully finish and return no data. Not sure if this is ideal, but works for now.